source/server/configuration_impl.h

```diff
@@ -56,6 +56,40 @@ class HttpTracerFactory {
 };
 
 /**
+ * Implemented for each Stats::Sink and registered via Registry::registerFactory() or
+ * the convenience class RegisterFactory.
+ */
+class StatsSinkFactory {
+public:
+  virtual ~StatsSinkFactory() {}
+
+  /**
+   * Create a particular Stats::Sink implementation. If the implementation is unable to produce a
+   * Stats::Sink with the provided parameters, it should throw an EnvoyException in the case of
+   * general error or a Json::Exception if the json configuration is erroneous. The returned
+   * pointer should always be valid.
+   * @param config supplies the custom proto configuration for the Stats::Sink
+   * @param server supplies the server instance
+   * @param cluster_manager supplies the cluster_manager instance
+   */
+  virtual Stats::SinkPtr createStatsSink(const Protobuf::Message& config, Instance& server,
+                                         Upstream::ClusterManager& cluster_manager) PURE;
```

Comment:
Why is `ClusterManager` a distinct parameter here? I think it can be inferred from the `Server::Instance` parameter?